### PR TITLE
[cpp] Allow changeJob binding to used on mobs

### PR DESCRIPTION
**Pull request by CatsEyeXI**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Allows changeJob() to be used on mobs. 
This functionality is needed to properly implement ENM's like "Pulling the Strings".

## Steps to test these changes
!zone Dragon's Aery
!getstats
Target a Spider
!exec target:changeJob(xi.job.PLD)
!getstats

### CODE REVIEW

File: src/map/lua/lua_baseentity.cpp [hunk 1]
```diff
@@ -5919,69 +5919,157 @@ uint8 CLuaBaseEntity::getSubJob()
 
 void CLuaBaseEntity::changeJob(uint8 newJob)
 {
-    auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
-
-    if (!PChar)
+    if (m_PBaseEntity && m_PBaseEntity->objtype == TYPE_PC)
```
Comment (zach2good) [inline src/map/lua/lua_baseentity.cpp]: `if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity)) ...`

File: src/map/lua/lua_baseentity.cpp [hunk 2]
```diff
@@ -5919,69 +5919,157 @@ uint8 CLuaBaseEntity::getSubJob()
...
+        PChar->pushPacket(new CCharRecastPacket(PChar));
+        PChar->pushPacket(new CCharAbilitiesPacket(PChar));
+        PChar->pushPacket(new CCharUpdatePacket(PChar));
+        PChar->pushPacket(new CMenuMeritPacket(PChar));
+        PChar->pushPacket(new CMonipulatorPacket1(PChar));
+        PChar->pushPacket(new CMonipulatorPacket2(PChar));
+        PChar->pushPacket(new CCharSyncPacket(PChar));
     }
-    else if (PChar->GetSJob() != JOB_BLU)
+    else
```
Comment (zach2good) [inline src/map/lua/lua_baseentity.cpp]: `else if (auto* PMob = dynamic_cast<CMobEntity*>(m_PBaseEntity)) ...`

File: src/map/lua/lua_baseentity.cpp [hunk 3]
```diff
@@ -5919,69 +5919,157 @@ uint8 CLuaBaseEntity::getSubJob()
...
+                PWeapon->setSkillType(SKILL_STAFF);
+                PWeapon->setDelay(8000);
+                PWeapon->setBaseDelay(8000);
+                break;
+            default:
+                break;
+        }
+
+        mobutils::CalculateMobStats(PMob);
+    }
```
Comment (zach2good) [inline src/map/lua/lua_baseentity.cpp:6051]: `else { Error condition logging here }`

File: src/map/lua/lua_baseentity.cpp [hunk 4]
```diff
@@ -5919,69 +5919,157 @@ uint8 CLuaBaseEntity::getSubJob()
...
+        PChar->pushPacket(new CCharJobsPacket(PChar));
+        PChar->pushPacket(new CCharStatsPacket(PChar));
+        PChar->pushPacket(new CCharSkillsPacket(PChar));
+        PChar->pushPacket(new CCharRecastPacket(PChar));
+        PChar->pushPacket(new CCharAbilitiesPacket(PChar));
+        PChar->pushPacket(new CCharUpdatePacket(PChar));
+        PChar->pushPacket(new CMenuMeritPacket(PChar));
+        PChar->pushPacket(new CMonipulatorPacket1(PChar));
+        PChar->pushPacket(new CMonipulatorPacket2(PChar));
+        PChar->pushPacket(new CCharSyncPacket(PChar));
```
Comment (zach2good) [inline src/map/lua/lua_baseentity.cpp]: Presumably these big chunks of logic are replicated from somewhere, so it makes sense to extract them into a `charutils` function. Same with the chunk below into `mobutils`

Comment (CatsEyeXI) [reply src/map/lua/lua_baseentity.cpp]: please bear with me as I'm learning. To be clear im just adding to the existing binding, so it isn't exactly duplicative, right? is the recommendation to move this entire logic into charutils?

Comment (zach2good) [reply src/map/lua/lua_baseentity.cpp]: Don't worry about it. Just do the recommended styling changes.

Comment (CatsEyeXI) [reply src/map/lua/lua_baseentity.cpp]: I think that's everything. Please let me know if I failed to make any of the changes as requested.

**Review by zach2good — APPROVED** [review]

**Pull request merged**